draft-ietf-taps-interface.md

```diff
@@ -1681,28 +1703,33 @@ when it is temporarily not ready to receive messages.
 Connection -> Received<messageData, messageContext>
 ~~~
 
-A Received event indicates the delivery of a complete Message. It contains two objects,
-the received bytes as messageData, and the metadata and properties of the received
-Message as messageContext. See {{receive-context}} for details about the received context.
+A Received event indicates the delivery of a complete Message. It contains two
+objects, the received bytes as messageData, and the metadata and properties of
+the received Message as messageContext.
+
+The messageData object provides access to the bytes that were received for this
+Message, along with the length of the byte array.
 
-The messageData object provides access to the bytes that were received for this Message,
-along with the length of the byte array.
+The messageContext is provided to retrieve metadata about the message, see
+{{msg-ctx}}, and to refer to the message later on, e.g., to enable to map
+responses to the message to the same underlying transport connection or stream,
+see {{send-replies}}.
 
-See {{receive-framing}} for handling Message framing in situations where the Protocol
-Stack provides octet-stream transport only.
+See {{receive-framing}} for handling Message framing in situations where the
+Protocol Stack provides octet-stream transport only.
 
 ### ReceivedPartial {#receive-partial}
 
 ~~~
-Connection -> ReceivedPartial<messageData, messageContext, endOfMessage>
+Connection -> ReceivedPartial<messageData, messageContext, messageContext, endOfMessage>
```

Comment:
huh? two messageContexts, with the same name even?

Reply:
rewriting artefact - fixed